lib/compress/zstd_opt.c

```diff
@@ -620,9 +667,10 @@ U32 ZSTD_insertBtAndGetAllMatches (
             matches[mnum].off = (current - matchIndex) + ZSTD_REP_MOVE;
             matches[mnum].len = (U32)matchLength;
             mnum++;
-            if (matchLength > ZSTD_OPT_NUM) break;
-            if (ip+matchLength == iLimit) {  /* equal : no way to know if inf or sup */
-                break;   /* drop, to preserve bt consistency (miss a little bit of compression) */
+            if ( (matchLength > ZSTD_OPT_NUM)
```

Comment:
@Cyan4973 should this be `sufficient_len` like above, to save a little bit of searches?

Reply:
Initially, the search used to stop on reaching `sufficient_len`.
But, if my memory serves me right, 
it was found that using `sufficient_len` as a search limit directly within the binary tree
was hurting compression performance too much.

This conclusion obviously depends on test sample,
but I kind of remember that some corner cases were significantly impacted,
with `btlazy2` achieving better performance just because of this.

The trade-off was to use `sufficient_len` mainly within the parser,
as opposed to the match finder.

Note though that `sufficient_len` is still used in hc3 and repcode finder,
probably because the conversion was half-done,
and the benchmark results were likely fixed just with this half.

This could be re-visited if need be.
This is part of the compression / speed trade-off territory.

However, it may take time to properly benchmark this scenario in order to argue that the new solution is better.